Sources/Web3Core/EthereumABI/ABIParsing.swift

```diff
@@ -7,16 +7,41 @@ import Foundation
 
 extension ABI {
 
-    public enum ParsingError: Swift.Error {
+    public enum ParsingError: LocalizedError {
         case invalidJsonFile
-        case elementTypeInvalid
+        case elementTypeInvalid(_ desc: String? = nil)
         case elementNameInvalid
         case functionInputInvalid
         case functionOutputInvalid
         case eventInputInvalid
         case parameterTypeInvalid
         case parameterTypeNotFound
         case abiInvalid
+
+        public var errorDescription: String? {
+            var errorMessage: [String?]
+            switch self {
+            case .invalidJsonFile:
+                errorMessage = ["invalidJsonFile"]
+            case .elementTypeInvalid(let desc):
+                errorMessage = ["elementTypeInvalid", desc]
```

Comment:
I wonder if this line is a cause that we're messing here with an array of Strings rather a plain String?